share/spice/timer/timer.js

```diff
@@ -7,290 +7,430 @@ License: CC BY-NC 3.0 http://creativecommons.org/licenses/by-nc/3.0/
+        Timer;
 
-    env.ddg_spice_timer = function(api_result) {
-        
-        //prevent flash of unstyled content if spice shows before css loads
-        function waitForCss() {
-            var interval = setInterval(findTimerCss, 10),
-                tries = 0;
-
-            function findTimerCss() {
-                var el = $('link[href$="timer.css"]')[0];
-                tries++;
-              
-                if (el) {
-                    clearInterval(interval);
-                    if (el.sheet) {
-                        //already loaded, no need for 'load' event
-                        showTimer();
-                    } else {
-                        //not yet loaded, attach a 'load' event
-                        $(el).load(showTimer);
-                    }
+    // helper methods
+
+    // add zeros to beginning of string
+    function padZeros(n, len) {
+        var s = n.toString();
+        while (s.length < len) {
+            s = '0' + s;
+        }
+        return s;
+    }
+
+    // split seconds into hours, minutes, seconds
+    function getHrsMinsSecs(totalSeconds) {
+        var result = {};
+
+        result.hours = Math.floor(totalSeconds / 3600);
+        totalSeconds %= 3600;
+        result.minutes = Math.floor(totalSeconds / 60);
+        result.seconds = Math.floor(totalSeconds % 60);
+
+        return result;
+    }
+
+    // get time in seconds from query
+    function parseQueryForTime() {
+        var q = DDG.get_query().replace('timer', '').replace('online', '').replace('s','sec').replace('m','min'),
+            regex = new RegExp(/([\d]+\.?[\d]*) ?(min|sec|h)/),
+            time = 0,
+            match,
+            val,
+            unit;
+
+        while (true) {
+            match = regex.exec(q);
+            if (match) {
+                val = parseFloat(match[1]);
+                unit = match[2];
+                if (unit === 'h') {
+                    time += val * 60 * 60;
                 }
-              
-                //give up after 30 seconds
-                else if (tries > 3000) {
-                    clearInterval(interval);
+                else if (unit === 'min') {
+                    time += val * 60;
                 }
+                else if (unit === 'sec') {
+                    val = Math.round(val);
+                    time += val;
+                }
+                q = q.replace(match[0], '');
+            } else {
+                break;
             }
         }
-      
-        waitForCss();
-        
-        function showTimer() {
-            $('#timer_container').removeClass('is-hidden');
+
+        return (time <= MAX_TIME) ? time : MAX_TIME;
+    }
+
+    //play the alarm sound
+    function playLoopingSound() {
+        function requirePlayer(player) {
+            player.play('alarm-sound', soundUrl, {
+                autoPlay: true
+            });
         }
+        DDG.require('audio', requirePlayer);
```

Comment:
@andrey-p I noticed on the live site (and here) the sound no longer works. Any ideas what's going on? I haven't investigated it yet.